### PR TITLE
feat: Support for building with Java 11 (SDKCF-4919)

### PR DESCRIPTION
**Pull request by leila-gu**
# Checklist
- [x] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [x] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data **before every commit**, including API endpoints and keys
- [x] I checked that all possible internal and platform/library exceptions are gracefully handled and **will not** crash the host app
- [x] I checked for open & known issues (especially crashes) for any newly added platform APIs or libraries
- [x] I tested non-trivial changes on a real device
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

File: analytics/build.gradle [hunk 1]
```diff
@@ -3,6 +3,6 @@ apply from: "../config/quality/gitleaks/build.gradle"
 apply plugin: 'kotlin-android'
 
 java {
-    sourceCompatibility = JavaVersion.VERSION_1_8
-    targetCompatibility = JavaVersion.VERSION_1_8
+    sourceCompatibility = JavaVersion.VERSION_11
```
Comment (vincentdelmundo) [inline analytics/build.gradle:6]: are these still necessary? I mean, does the SDK still need to manually set the compatibility even if the `config` already set it to Java 11?

**Review by vincentdelmundo — CHANGES_REQUESTED** [review]
Looks good. Just a few comments
* please make sure that documentation can be generated correctly
* update the java version in bitrise

File: analytics/build.gradle (hunk 1 above)
Comment (leila-gu) [reply analytics/build.gradle]: You are right, only Kotlin JvmTarget needs the manually set. Updated.

**Comment by leila-gu** [comment]
> 

Bitrise config updated
The documentation is generated correctly.

File: inappmessaging/USERGUIDE.md [hunk 2]
```diff
@@ -388,6 +388,9 @@ Documents targeting Product Managers:
 ## <a name="changelog"></a> Changelog
 
 ### 7.0.0 (in-progress)
+* SDKCF-4919: **Breaking Changes:** Build using Java 11.
```
Comment (vincentdelmundo) [inline inappmessaging/USERGUIDE.md:391]: Please check if host app can no longer use our SDK if the app does not use Java 11

**Review by vincentdelmundo — APPROVED** [review]
LGTM

File: analytics/build.gradle (hunk 1 above)
Comment (leila-gu) [reply analytics/build.gradle]: The app can still run and receive the in-app messages when compiling/targeting java 8. I removed the Breaking change note.

File: inappmessaging/USERGUIDE.md (hunk 2 above)
Comment (leila-gu) [reply inappmessaging/USERGUIDE.md]: The app can still run and receive the in-app messages when compiling/targeting java 8. I removed the Breaking change note.

**Pull request merged**